File: tests/integration/construct_component_methods_unknown/nodejs/component.ts
// Copyright 2016-2021, Pulumi Corporation.  All rights reserved.

import * as pulumi from "@pulumi/pulumi";

export class Component extends pulumi.ComponentResource {
    constructor(name: string, opts?: pulumi.ComponentResourceOptions) {
        super("testcomponent:index:Component", name, undefined, opts, true);
    }

    getMessage(args: Component.GetMessageArgs): pulumi.Output<Component.GetMessageResult> {
        return pulumi.runtime.call("testcomponent:index:Component/getMessage", {
            "__self__": this,
            "echo": args.echo,
        }, this);
    }
}

export namespace Component {
    export interface GetMessageArgs {
        echo: pulumi.Input<string>;
    }

    export interface GetMessageResult {
        message: string;
    }
}

File: tests/integration/construct_component_methods_unknown/dotnet/Component.cs
// Copyright 2016-2021, Pulumi Corporation.  All rights reserved.

using Pulumi;

class Component : ComponentResource
{
    public Component(string name, ComponentResourceOptions? opts = null)
        : base("testcomponent:index:Component", name, ResourceArgs.Empty, opts, remote: true)
    {
    }

    public Output<ComponentGetMessageResult> GetMessage(ComponentGetMessageArgs args)
        => Deployment.Instance.Call<ComponentGetMessageResult>("testcomponent:index:Component/getMessage", args, this);
}

public class ComponentGetMessageArgs : CallArgs
{
    [Input("echo")]
    public Input<string> Echo { get; set; } = null!;
}

[OutputType]
public sealed class ComponentGetMessageResult
{
    public readonly string Message;

    [OutputConstructor]
    private ComponentGetMessageResult(string message)
    {
        Message = message;
    }
}
